core/shared/src/main/scala/chartreuse/Plot.scala

```diff
@@ -50,19 +54,196 @@ final case class Plot[
     copy(yTitle = newYTitle)
   }
 
-  //  TODO: handle axes and grid layout
-  def draw(width: Int, height: Int): Picture[
-    Alg & Text & doodle.algebra.Transform & Debug,
-    Unit
-  ] = {
-    val plot =
+  def withGrid(newGrid: Boolean): Plot[A, Alg] = {
+    copy(grid = newGrid)
+  }
+
+  def draw(width: Int, height: Int): PlotPicture = {
+    val dataBoundingBox = layers.foldLeft(BoundingBox.empty) { (bb, layer) =>
+      bb.on(layer.data.boundingBox(layers.head.toPoint))
```

Comment:
`layers.head.toPoint` is wrong. Different layers could have different `toPoint` functions. This only compiles due to the overly restrictive type of layers (should probably be `Layer[?, Alg]` not `Layer[A, Alg]`)

```suggestion
      bb.on(layer.data.boundingBox(layer.toPoint))
```

It might be worth adding a `boundingBox` method to `Layer`.